wagtail/admin/templates/wagtailadmin/shared/side_panels/includes/status/locale.html

```diff
@@ -8,15 +8,15 @@
         {% trans 'Locale: ' as screen_reader_title_prefix %}
     {% endif %}
 
-    {% if translations %}
+    {% if object.pk and translations %}
         {% blocktrans trimmed asvar help_text %}
             Available in {{ translations_total }} locales
         {% endblocktrans %}
     {% else %}
         {% trans 'No other translations' as help_text %}
     {% endif %}
 
-    {% with icon_name='globe' title=object.locale.get_display_name %}
+    {% with icon_name='globe' title=locale.get_display_name %}
```

Comment:
The object's locale is still unset in initial load of the page for snippets. Using the `locale` context should work fine for both pages and snippets.